utils/set-alert-policy-required-datasources.ts

```diff
@@ -0,0 +1,126 @@
+import { passedProcessArguments } from './lib/helpers';
+import { CUSTOM_EVENT, recordNerdGraphResponse } from './newrelic/customEvent';
+import {
+  fetchPaginatedGHResults,
+  filterQuickstartConfigFiles,
+  isNotRemoved,
+} from './lib/github-api-helpers';
+import Quickstart from './lib/Quickstart';
+import Alert, { SubmitSetRequiredDataSourcesMutationResult } from './lib/Alert';
+import { chunk } from './lib/nr-graphql-helpers';
+
+type QuickstartResult = {
+  name: string;
+  dataSourceIds: string[];
+};
+
+const getQuickstartNameAndDataSources = async (
+  ghUrl?: string,
+  ghToken?: string
+): Promise<{ hasFailed: boolean; results: QuickstartResult[] }> => {
+  if (!ghToken) {
+    console.error('GITHUB_TOKEN is not defined.');
+    return { hasFailed: true, results: [] };
+  }
+
+  if (!ghUrl) {
+    console.error('Github PR URL is not defined.');
+    return { hasFailed: true, results: [] };
+  }
+
+  const files = await fetchPaginatedGHResults(ghUrl, ghToken);
+
+  const quickstartNames = filterQuickstartConfigFiles(files)
+    .filter(isNotRemoved)
+    .map(({ filename }) => {
+      const quickstart = new Quickstart(filename);
+      return {
+        name: quickstart.config.title,
+        dataSourceIds: quickstart.config.dataSourceIds ?? [],
+      };
+    });
+
+  return { hasFailed: false, results: quickstartNames };
+};
+
+const setAlertPoliciesRequiredDataSources = async (
+  ghUrl: string,
+  ghToken: string | undefined
+): Promise<boolean> => {
+  const {
+    hasFailed: hasQuickstartNamesFailed,
+    results: quickstartNamesAndDataSources,
+  } = await getQuickstartNameAndDataSources(ghUrl, ghToken);
+
+  if (hasQuickstartNamesFailed) {
+    return hasQuickstartNamesFailed;
+  }
+
+  let results: SubmitSetRequiredDataSourcesMutationResult[] = [];
+
+  for (const c of chunk(Object.entries(quickstartNamesAndDataSources), 5)) {
+    const result = await Promise.all(
+      c.map(async ([_, quickstart]) => {
+        const alertPolicy = await Alert.getAlertPolicyRequiredDataSources(
```

Comment:
[part 2] right before the request is make, we should do 
```tsx 
const hasFailed = quickstart.dataSourceIds.length > 1;
```
and then return errors before attempting to make the request